postgresql/files/restore.sh

```diff
@@ -33,11 +33,20 @@ touch /root/postgresql/flags/{{ database.name }}-installed
 {%- set dest_name = db_name + ".pg_dump.gz" %}
 {%- set target = "/root/postgresql/data/" %}
 
-scp backupninja@{{ database.initial_data.source }}:/srv/backupninja/{{ host }}/var/backups/postgresql/postgresql.{{ age }}/{{ source_name }} {{ target }}{{ dest_name }} 
+{%- if database.initial_data.auth is defined %}
+export AWS_ACCESS_KEY_ID={{ database.initial_data.auth.awsaccesskeyid }}
+export AWS_SECRET_ACCESS_KEY={{ database.initial_data.auth.awssecretaccesskey }}
+{%- endif %}
 
+{%- if database.initial_data.engine == "backupninja" %}
+scp backupninja@{{ database.initial_data.source }}:/srv/backupninja/{{ host }}/var/backups/postgresql/postgresql.{{ age }}/{{ source_name }} {{ target }}{{ dest_name }} 
 gunzip -d -1 -f {{ target }}{{ dest_name }}
-
 pg_restore /root/postgresql/data/{{ db_name }}.pg_dump --dbname={{ db_name }} --host=localhost --username={{ database.users[0].name }} --no-password -c
+{%- else %}
+duplicity --s3-unencrypted-connection --s3-use-new-style --s3-european-buckets s3+http://{{ database.initial_data.source }}/{{ host }} {{ target }}{{ dest_name }} 
+gunzip -d -1 -f {{ target }}{{ dest_name }}/var/backups/postgresql/{{ source_name }}
+pg_restore /root/postgresql/data/{{ dest_name }}/var/backups/postgresql/{{ db_name }}.pg_dump --dbname={{ db_name }} --host=localhost --username={{ database.users[0].name }} --no-password -c
+{%- endif %}
 
 touch /root/postgresql/flags/{{ db_name }}-installed
```

Comment:
This is pretty poor way.
- better to use "postgresql" user directory.
- better to include timestamp

Before file is created, validate db with:
https://serverfault.com/questions/175865/how-to-check-postgresql-database-integrity